backend/pkg/config/kafka_sasl_oauth.go

```diff
@@ -10,25 +10,75 @@
 package config
 
 import (
+	"context"
+	"encoding/base64"
 	"flag"
 	"fmt"
+	"io"
+	"net/http"
 )
 
 // KafkaSASLOAuthBearer is the config struct for the SASL OAuthBearer mechanism
 type KafkaSASLOAuthBearer struct {
-	Token string `yaml:"token"`
+	Token         string `yaml:"token"`
+	ClientID      string `yaml:"clientid"`
+	ClientSecret  string `yaml:"clientsecret"`
+	TokenEndpoint string `yaml:"tokenendpoint"`
 }
 
 // RegisterFlags registers all sensitive Kerberos settings as flag
 func (c *KafkaSASLOAuthBearer) RegisterFlags(f *flag.FlagSet) {
 	f.StringVar(&c.Token, "kafka.sasl.oauth.token", "", "OAuth Bearer Token")
+	f.StringVar(&c.ClientID, "kafka.sasl.oauth.clientid", "", "OAuth Bearer Client Id")
+	f.StringVar(&c.ClientSecret, "kafka.sasl.oauth.clientsecret", "", "OAuth Bearer Client Secret")
+	f.StringVar(&c.TokenEndpoint, "kafka.sasl.oauth.tokenendpoint", "", "OAuth Bearer Token Endpoint")
```

Comment:
```suggestion
	f.StringVar(&c.TokenEndpoint, "kafka.sasl.oauth.token-endpoint", "", "OAuth Bearer Token Endpoint")
```

Reply:
I have reverted your suggestion, as I could not get it to work with ENV variable injection.
 `KAFKA_SASL_OAUTH_TOKEN-ENDPOINT` and `KAFKA_SASL_OAUTH_TOKEN_ENDPOINT` did not work.

Keeping it as `tokenendpoint` allowed me to set `KAFKA_SASL_OAUTH_TOKENENDPOINT=`

Reply:
The flags are entirely separate from the env variables. The flag name should not have any impact on the env variable name, it's derived from the struct tags (`yaml`).